components/brave_extension/extension/brave_extension/background/webDiscoveryProject.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import { App } from '../web-discovery-project/build'
+
+if (App !== undefined) {
+  const APP = new App({
+    version: chrome.runtime.getManifest().version,
+  });
+
+  APP.start().then(() => {
+    const toggleWebDiscoveryExtension = (enabled: boolean) => {
+      APP.prefs.set('modules.web-discovery-project.enabled', enabled)
+      APP.prefs.set('modules.hpnv2.enabled', enabled);
+    }
```

Comment:
I am wondering if in this case we could actually stop the whole App using `APP.stop()`.

Reply:
Now that I think about it, maybe we can also initialize the `App` only if the user opted-in?
```js
chrome.settingsPrivate.getPref(WEB_DISCOVERY_PREF_KEY, (pref) => {
  if (pref && pref.type === chrome.settingsPrivate.PrefType.BOOLEAN) {
    APP.start().then(...);
    }
});
```

I guess we could also lazily instantiate the `App` (i.e. `const APP = new App(...)`) when the pref is set so that this code does nothing unless the user opted-in.

Reply:
i changed it to do this